lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper.rb

```diff
@@ -41,55 +76,113 @@ def self.calc_prev_release_version(version)
           "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}"
         end
   
+        # Compute the name of the next build version.
+        #
+        # @param [String] version The current version we want to increment
+        #
+        # @return [String] The predicted next build version, in the form of "X.Y.Z.N".
+        #         Corresponds to incrementing the last (4th) component N of the version.
+        #
         def self.calc_next_build_version(version)
           vp = get_version_parts(version)
           vp[BUILD_NUMBER] += 1
           "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}.#{vp[HOTFIX_NUMBER]}.#{vp[BUILD_NUMBER]}"
         end
   
+        # Compute the name of the next hotfix version.
+        #
+        # @param [String] version The current version we want to increment
+        #
+        # @return [String] The predicted next hotfix version, in the form of "X.Y.Z".
+        #         Corresponds to incrementing the 3rd component of the version.
+        #
         def self.calc_next_hotfix_version(version)
           vp = get_version_parts(version)
           vp[HOTFIX_NUMBER] += 1
           "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}.#{vp[HOTFIX_NUMBER]}"
         end
   
+        # Compute the name of the previous build version.
+        #
+        # @param [String] version The current version we want to decrement
+        #
+        # @return [String] The predicted previous build version, in the form of "X.Y.Z.N".
+        #         Corresponds to decrementing the last (4th) component N of the version.
+        #
         def self.calc_prev_build_version(version)
           vp = get_version_parts(version)
           vp[BUILD_NUMBER] -= 1 unless vp[BUILD_NUMBER] == 0
           "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}.#{vp[HOTFIX_NUMBER]}.#{vp[BUILD_NUMBER]}"
         end
   
+        # Compute the name of the previous hotfix version.
+        #
+        # @param [String] version The current version we want to decrement
+        #
+        # @return [String] The predicted previous hotfix version, in the form of "X.Y.Z", or "X.Y" if Z is 0.
+        #         Corresponds to decrementing the 3rd component Z of the version, striping it if it ends up being zero.
+        #
         def self.calc_prev_hotfix_version(version)
           vp = get_version_parts(version)
           vp[HOTFIX_NUMBER] -= 1 unless vp[HOTFIX_NUMBER] == 0
           return "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}.#{vp[HOTFIX_NUMBER]}" unless vp[HOTFIX_NUMBER] == 0
           "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}"
         end
 
+        # Create an internal version number, for which the build number is based on today's date.
+        #
+        # @param [String] version The current version to create an internal version name for.
+        #
+        # @return [String] The internal version, in the form of "X.Y.Z.YYYYMMDD".
```

Comment:
Not necessarily important to _this_ PR, but this assumes that we never have more than one beta in a day. Regrettably, this assumption is incorrect 😅

Might be worth addressing down the road?

Reply:
https://github.com/wordpress-mobile/release-toolkit/issues/202